spring-cloud-gcp-data-firestore/src/test/java/org/springframework/cloud/gcp/data/firestore/FirestoreCiTests.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2019-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.firestore;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+
+import ch.qos.logback.classic.Level;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.firestore.PublicClassMapper;
+import com.google.firestore.v1.CreateDocumentRequest;
+import com.google.firestore.v1.Document;
+import com.google.firestore.v1.FirestoreGrpc;
+import com.google.firestore.v1.Value;
+import io.grpc.CallCredentials;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.auth.MoreCallCredentials;
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.LoggerFactory;
+
+import org.springframework.cloud.gcp.data.firestore.util.ObservableReactiveUtil;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assume.assumeThat;
+
+/**
+ * @author Dmitry Solomakha
+ */
+public class FirestoreCiTests {
+	private static final String DEFAULT_PARENT = "projects/spring-firestore-native-test/databases/(default)/documents";
+
+	private FirestoreGrpc.FirestoreStub firestoreStub;
+
+	private FirestoreTemplate firestoreTemplate;
+
+
+	@Before
+	public void setup() throws IOException {
+
+		assumeThat(
+				"Firestore-sample tests are disabled. Please use '-Dit.firestore=true' "
+						+ "to enable them. ",
+				System.getProperty("it.firestore"), is("true"));
```

Comment:
Also need to modify the travis file to pass in -Dit.firestore for tests to be run.